.github/workflows/publish-charts.yaml

```diff
@@ -0,0 +1,88 @@
+name: publish-charts
+
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Repository from where we want to build Charts'
+        required: true
+      repo_ref:
+        description: 'The branch/tag that will be used as new version for charts. Usually tag ref'
+        required: true
+      repo_chart:
+        description: 'Object with source path, destination path and chart name'
+        required: true
+      currentChartVersion:
+        description: 'Chart version set in `Chart.yaml`. Default: `LATEST`'
+        required: false
+        default: 'LATEST'
+
+env:
+  REPO: ${{ github.event.inputs.repo }}
+  REPO_REF: ${{ github.event.inputs.repo_ref }}
+  CurrentChartVersion: ${{ github.event.inputs.currentChartVersion }}
+
+jobs:
+  publish-charts:
+    runs-on: ubuntu-latest
+    name: Publish charts
+    continue-on-error: true
+    strategy:
+      max-parallel: 1
+      matrix: ${{ fromJson(github.event.inputs.repo_chart) }}
+    steps:
+
+      - id: checkout-gh-pages
+        uses: actions/checkout@v2 # Checkout Charts - gh-pages
+        with:
+          ref: 'gh-pages'
+          fetch-depth: 0
+
+      - id: checkout-helm-src
+        uses: actions/checkout@v2 # Checkout Helm source code
+        with:
+          repository: ${{ env.REPO }}
+          ref: ${{ env.REPO_REF }}
+          path: ${{ env.REPO }}
+
+      - id: chart-version
+        name: 'Set version - ${{ env.REPO_REF }}'
+        if: ${{ steps.checkout-gh-pages.outcome == 'success' && steps.checkout-helm-src.outcome == 'success' }}
+        run: |
+          find . \( -name "Chart.yaml" -o -name "values.yaml" \) -exec sed -i s/${{ env.CurrentChartVersion }}/${{ env.REPO_REF }}/ {} + 
+          echo "Replacing ${{ env.CurrentChartVersion }} with new version ${{ env.REPO_REF }}."
+        working-directory: ./${{ env.REPO }}
+
+      - id: check
+        name: 'Check if chart already exist'
+        run: |
+          FILE=$(grep name ${{ env.REPO }}/${{ matrix.src_path }}Chart.yaml | awk '{print $2}')
```

Comment:
I am thinking... Can this get messed up somehow? What if `${{ matrix.src_path }}` without an ending slash? eg: `my-chart/foo`. If so, i see two options:
- append `/` if it is missing at the end of `${{ matrix.src_path }}`
- rewrite this expression to: `${{ env.REPO }}/${{ matrix.src_path }}/Chart.yaml`. if `${{ matrix.src_path }}` does contain `/`, remove it prior to evaluating the expression above

@ljubon what do you think?

Reply:
yep, actually, Dusan commented the same thing previously so I've put `NOTE` in [PR description](https://github.com/G-Research/charts/pull/51#issue-973085674) 
whatever we do, still someone can forget `/` at the end of the string OR add it unintentionally and still we can't predict either action. 
So I decided at that time to have example `input` where I added a note that we MUST have `/` at the end of `src_path` and i think also `dst_path`. So it would be clear for them that it will not work if they do it differently.